google-cloud-resourcemanager/src/main/java/com/google/cloud/resourcemanager/spi/v1beta1/HttpResourceManagerRpc.java

```diff
@@ -16,58 +16,168 @@
+      ImmutableMap.<Integer, Integer>builder()
+          .put(0, 200)
+          .put(1, 499)
+          .put(2, 500)
+          .put(3, 400)
+          .put(4, 504)
+          .put(5, 404)
+          .put(6, 409)
+          .put(7, 403)
+          .put(16, 401)
+          .put(8, 429)
+          .put(9, 400)
+          .put(10, 409)
+          .put(11, 400)
+          .put(12, 501)
+          .put(13, 500)
+          .put(14, 503)
+          .put(15, 500)
+          .build();
+
+  private static final ResultRetryAlgorithm<Operation> OPERATION_HANDLER =
+      new ResultRetryAlgorithm<Operation>() {
+        @Override
+        public TimedAttemptSettings createNextAttempt(
+            Throwable prevThrowable, Operation prevResponse, TimedAttemptSettings prevSettings) {
+          return null;
+        }
+
+        @Override
+        public boolean shouldRetry(Throwable prevThrowable, Operation prevOp)
+            throws CancellationException {
+          return prevThrowable != null || prevOp.getDone() == null || !prevOp.getDone();
+        }
+      };
+
+  private final CloudResourceManager resourceManager;
+  private final ApiClock clock;
 
   public HttpResourceManagerRpc(ResourceManagerOptions options) {
     HttpTransportOptions transportOptions = (HttpTransportOptions) options.getTransportOptions();
     HttpTransport transport = transportOptions.getHttpTransportFactory().create();
     HttpRequestInitializer initializer = transportOptions.getHttpRequestInitializer(options);
     resourceManager =
-        new Cloudresourcemanager.Builder(transport, new JacksonFactory(), initializer)
+        new CloudResourceManager.Builder(transport, new JacksonFactory(), initializer)
             .setRootUrl(options.getHost())
             .setApplicationName(options.getApplicationName())
             .build();
+    clock = options.getClock();
   }
 
   private static ResourceManagerException translate(IOException exception) {
     return new ResourceManagerException(exception);
   }
 
+  private static ResourceManagerException translate(Status status) {
+    int code;
+    if (RPC_TO_HTTP_CODES.containsKey(status.getCode())) {
+      code = RPC_TO_HTTP_CODES.get(status.getCode());
+    } else {
+      code = BaseHttpServiceException.UNKNOWN_CODE;
+    }
+    return new ResourceManagerException(code, status.getMessage());
+  }
+
   @Override
   public Project create(Project project) {
     try {
-      return resourceManager.projects().create(project).execute();
+      final Operation operation = resourceManager.projects().create(project).execute();
+
+      Operation finishedOp =
+          runWithRetries(
+              new Callable<Operation>() {
+                @Override
+                public Operation call() {
+                  try {
+                    return resourceManager.operations().get(operation.getName()).execute();
+                  } catch (IOException ex) {
+                    throw translate(ex);
```

Comment:
In case of error you this exception will be wrapped again on line 166, so wrapping/translating it here seems redundant.

Reply:
I don't think it will. This line deals with `get` throwing, but L166 deals with `get` itself succeeding but reporting that the operation wasn't successful right?